tools/deviceGen/output/device_capabilities_traits_defaults.hlsl

```diff
@@ -0,0 +1,511 @@
+NBL_GENERATE_GET_OR_DEFAULT(bufferImageGranularity, uint32_t, 1u << 16);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorSamplers, uint32_t, 16);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUBOs, uint32_t, 15);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorSSBOs, uint32_t, 31);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorImages, uint32_t, 96);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorStorageImages, uint32_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorInputAttachments, uint32_t, 7);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageResources, uint32_t, 127);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetSamplers, uint32_t, 80);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUBOs, uint32_t, 90);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetDynamicOffsetUBOs, uint32_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetSSBOs, uint32_t, 155);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetDynamicOffsetSSBOs, uint32_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetImages, uint32_t, 480);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetStorageImages, uint32_t, 40);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetInputAttachments, uint32_t, 7);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxVertexOutputComponents, uint16_t, 124);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationGenerationLevel, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationPatchSize, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlPerVertexInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlPerVertexOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlPerPatchOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlTotalOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationEvaluationInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationEvaluationOutputComponents, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryShaderInvocations, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryOutputVertices, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryTotalOutputComponents, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentInputComponents, uint32_t, 116);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentOutputAttachments, uint32_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentDualSrcAttachments, uint32_t, 1);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentCombinedOutputResources, uint32_t, 16);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeSharedMemorySize, uint32_t, 1 << 15);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCountX, uint32_t, MinMaxWorkgroupCount);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCountY, uint32_t, MinMaxWorkgroupCount);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCountZ, uint32_t, MinMaxWorkgroupCoun);
```

Comment:
how does it happen that your script generates missing `t` do you later go over the files by hand?

Reply:
hmm you have this in `maxViewportDimsY` too